packages/front-end/components/Settings/NewDataSourceForm.tsx

```diff
@@ -268,7 +268,10 @@ const NewDataSourceForm: FC<{
 
     const newVal = {
       ...datasource,
-      settings,
+      settings: {
+        ...settings,
+        schemaOptions: form.watch("settings.schemaOptions"),
```

Comment:
With this change, anytime we're working schemaFormat that contains `schemaOptions` the `schemaOptions` will be persisted on the datasource in the DB. 

From what I can tell, we haven't needed to do this previously because the values here were only used on the initial default query generation, and never again, But now, in Amplitude's case, we need to persist the `projectId` for future use in the auto metrics queries.

Open to suggestions on alternatives.